homeassistant/core.py

```diff
@@ -17,19 +18,7 @@
 import threading
 from time import monotonic
 from types import MappingProxyType
-from typing import (
-    TYPE_CHECKING,
-    Any,
-    Awaitable,
-    Callable,
-    Collection,
-    Coroutine,
-    Iterable,
-    Mapping,
-    Optional,
-    TypeVar,
-    cast,
-)
+from typing import TYPE_CHECKING, Any, Callable, Coroutine, Optional, TypeVar, cast
```

Comment:
We can move `Coroutine` too, right?

Reply:
Unfortunately we can't. Technically all the types from `collections.abc` are only subscriptable starting with Python 3.9, which results in RuntimeErrors for PY38. However with postponed evaluation of type annotations (`from __future__ import annotations`) they can be used in most cases.

The notable exceptions are type assignments `Alias = typing.List[int]` and `cast(typing.List[int], var)`. Without the `pylint` checker, I wouldn't have changed anything, since it's just too complicated to figure out if an import can safely be replaced manually.

I'm pretty confident in the current change though. `pylint` already has some good checks for invalid PEP 585 and 604 syntax.

Reply:
We're moving `Coroutine` in some other places. What's the distinction? Is it that we use the type in a `cast` call here that wouldn't work?

Reply:
Exactly, everything that is evaluated at runtime requires `3.9`
IMO it's probably not worth to think about it in detail. I frequently don't know if something is possible and even other code checkers  (like `pylance`) don't get everything right, so I would suggest to just use what works. Everything else regarding type aliases and the alternative union syntax will soon be handled by `pylint`.

Reply:
> But in this particular case `Coroutine` is used in `cast` without specified types (non subscriptable), so it should be possible to replace it with `collections.abc`.

At the moment, I would recommend not to change it though. Someone only needs to add a specific type and it won't work. Better be safe here.

Reply:
Yeah, I think such cases can be fixed later when 3.9 is the default or when can tooling like `pylint` can verify this automatically.

Reply:
Ok.